rust/connlib/libs/tunnel/src/peer_handler.rs

```diff
@@ -16,9 +16,11 @@ where
 {
     #[inline(always)]
     fn is_wireguard_packet_ok(&self, parsed_packet: &Packet, peer: &Peer) -> bool {
+        use secrecy::ExposeSecret;
+
         match &parsed_packet {
             Packet::HandshakeInit(p) => {
-                parse_handshake_anon(&self.private_key, &self.public_key, p).is_ok()
+                parse_handshake_anon(self.private_key.expose_secret(), &self.public_key, p).is_ok()
```

Comment:
@conectado Any thoughts on whether this will affect performance? My understanding is that `secrecy` mainly focuses on build time guarantees and so this change shouldn't affect memory access characteristics at all.

Reply:
As explained above this isn't needed.